solutions/CSES Sliding Cost.mdx

```diff
@@ -0,0 +1,100 @@
+---
+id: cses-1077
+source: CSES
+title: Sliding Cost
+author: Isaac Noel
+---
+
+**Time Complexity:** $\mathcal{O}(N\log K)$
+
+## Abstract
+
+Finding the difference between sums of the upper $K/2$ elements and lower $K/2$ with two multisets.
+
+<Warning>
+
+The implementation of this solution assumes familarity with [Sliding Median](/gold/sliding#problem-https://cses.fi/problemset/task/1076) as their implementations are near identical.
+
+</Warning>
+
+## Solution
+
+### The cost of a window
+
+It can be shown that it is optimal to change the values of the window to the median (left as an exercise :D). Once we find the median, we must find the sum of all elements' distance from the median. Rather than finding the absolute value of every individual element's distance, which is far too slow, we can split the window into two groups and use $\mathcal{O}(\log K)$ insertions/deletions to find the cost in constant time. The smallest $K/2$ elements in the window will be in the lower group while the largest $K/2$ elements in the window will be in the upper group.
```

Comment:
> we can split the window into two groups and use $\mathcal{O}(\log K)$ insertions/deletions to find the cost in constant time.

seems confusing. Do you mean that a single insertion runs in $\mathcal{O}(\log K)$  time? That's not constant time.

Reply:
>  all elements' distance

the distances of all elements?

Reply:
> > we can split the window into two groups and use $\mathcal{O}(\log K)$ insertions/deletions to find the cost in constant time.
> 
> seems confusing. Do you mean that a single insertion runs in $\mathcal{O}(\log K)$ time? That's not constant time.

Yeah I meant that an insertion runs in $\mathcal{O}(\log K)$ time which allows for the isolated act of finding the cost ($S_2-S_1$) to run in constant time. I'll try to make this more clear